src/index.ts

```diff
@@ -30,38 +28,21 @@ const logger = createLogger('Application');
 export const PROVIDER_STALL_TIMEOUT = 750;
 export const PROVIDER_DEFAULT_WEIGHT = 1;
 
-export const getProvider = async (
-  config: KeeperConfig['providerApiKeys'],
-  network: Network
-): Promise<providers.FallbackProvider> => {
-  // Infura has the highest priority (indicated by the lowest priority number).
-  const providersConfig: providers.FallbackProviderConfig[] = [
-    {
-      provider: new providers.InfuraProvider(network, config.infura),
-      priority: 10,
-      stallTimeout: PROVIDER_STALL_TIMEOUT,
-      weight: PROVIDER_DEFAULT_WEIGHT,
-    },
-  ];
+export const getProvider = (config: KeeperConfig['providerUrls']): providers.JsonRpcProvider => {
   if (config.alchemy) {
-    logger.info('Alchemy API key provided. Adding as fallback provider');
-    providersConfig.push({
-      provider: new providers.AlchemyProvider(network, config.alchemy),
-      priority: 20,
-      stallTimeout: PROVIDER_STALL_TIMEOUT,
-      weight: PROVIDER_DEFAULT_WEIGHT,
-    });
+    return new providers.JsonRpcProvider(config.alchemy);
```

Comment:
Before we had infura as the highest priority, just making sure it's a deliberate change to use alchemy instead

Reply:
Semi intentional 😓. I'll add a round robin style provider to compensate for this at some point. Something like this:

```js
export class ProviderCapsule {
  private readonly wallet: Wallet;
  private readonly providerRoundRobin: Generator<providers.BaseProvider, void, unknown>;
  private readonly logger: Logger;

  constructor(
    ethHdwalletMnemonic: string,
    network: Network,
    providerApiKeys: KeeperConfig['providerApiKeys'],
    providerUrls: KeeperConfig['providerUrls'],
    providerGlobalReqTimeout: number
  ) {
    axios.defaults.timeout = providerGlobalReqTimeout;

    const _providers: providers.BaseProvider[] = [];
    if (providerApiKeys.infura) {
      _providers.push(new providers.InfuraProvider(network, providerApiKeys.infura));
    }
    if (providerApiKeys.alchemy) {
      _providers.push(new providers.AlchemyProvider(network, providerApiKeys.alchemy));
    }
    if (providerUrls.quicknode) {
      _providers.push(new providers.JsonRpcProvider(providerUrls.quicknode));
    }
    this.wallet = Wallet.fromMnemonic(ethHdwalletMnemonic);
    this.providerRoundRobin = getRoundRobinGenerator(_providers);
    this.logger = createLogger('ProviderCapsule');

    this.logger.info('Registered RPC providers', {
      args: { n: _providers.length, providers: _providers.map(p => p.constructor.name).join(',') },
    });
  }

  get provider(): providers.BaseProvider {
    return this.providerRoundRobin.next().value!;
  }

  get signer(): Signer {
    return this.wallet.connect(this.provider);
  }
}
```